### PR TITLE
Clean up ShieldDur && ShieldDur

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -3734,45 +3734,31 @@ int __fastcall PM_DoRangeAttack(int pnum)
...
-	int v4; // ecx
-	int v5; // ecx
-	int v6; // ecx
-
-	v1 = pnum;
-	if ( pnum == myplr )
-	{
-		if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( pnum == myplr ) {
+		if ( (unsigned int)pnum >= MAX_PLRS ) {
```
Comment (seritools) [inline Source/player.cpp]: `(DWORD)`

File: Source/player.cpp [hunk 2]
```diff
@@ -3734,45 +3734,31 @@ int __fastcall PM_DoRangeAttack(int pnum)
...
 			TermMsg("ShieldDur: illegal player %d", pnum);
-		v2 = v1;
-		if ( plr[v1].InvBody[4]._itype == ITYPE_SHIELD )
-		{
-			v3 = plr[v2].InvBody[4]._iDurability;
-			if ( v3 == 255 )
+		}
+
+		if ( plr[pnum].InvBody[4]._itype == ITYPE_SHIELD ) {
```
Comment (seritools) [inline Source/player.cpp]: I've added the `inv_body_loc` enum for InvBody indices

File: Source/player.cpp [hunk 3]
```diff
@@ -3734,45 +3734,31 @@ int __fastcall PM_DoRangeAttack(int pnum)
...
 				return;
-			v4 = v3 - 1;
-			plr[v2].InvBody[4]._iDurability = v4;
-			if ( !v4 )
-			{
+			}
+
+			plr[pnum].InvBody[4]._iDurability--;
+			if ( !plr[pnum].InvBody[4]._iDurability ) {
 				NetSendCmdDelItem(1u, 4u);
```
Comment (seritools) [inline Source/player.cpp]: Remove all the `u` suffixes if possible

File: Source/player.cpp [hunk 4]
```diff
@@ -3781,22 +3767,25 @@ void __fastcall ShieldDur(int pnum)
...
+	}
+
+	if ( plr[pnum]._pAnimFrame >= plr[pnum]._pBFrames ) {
+		StartStand(pnum, plr[pnum]._pdir);
+		ClearPlrPVars(pnum);
+
+		if ( !random(3, 10) ) {
+			ShieldDur(pnum);
+		}
+		return 1;
```
Comment (seritools) [inline Source/player.cpp]: This looks like a `BOOL` function to me :)

File: Source/player.cpp [hunk 5]
```diff
@@ -3734,69 +3734,58 @@ int __fastcall PM_DoRangeAttack(int pnum)
...
-	int v4; // ecx
-	int v5; // ecx
-	int v6; // ecx
-
-	v1 = pnum;
-	if ( pnum == myplr )
-	{
-		if ( (unsigned int)pnum >= MAX_PLRS )
+	if ( pnum == myplr ) {
+		if ( (DWORD)pnum >= MAX_PLRS ) {
```
Comment (nomdenom) [inline Source/player.cpp]: Looks like this `if` statement can never be true, probably a "bug" (oversight) in the original code :)

Comment (seritools) [reply Source/player.cpp]: They probably just spammed asserts everywhere pnum is used :D

Comment (mewmew) [reply Source/player.cpp]: Haha, yea. Good catch. Took me a while to see it. But pnum == myplr enforces that pnum is [0, 3].

Comment (AJenbo) [reply Source/player.cpp]: I copy pasted the clean up from another function so the tradition continues :P

**Pull request merged**